be2-scala/src/main/scala/ch/epfl/pop/authentication/Authenticate.scala

```diff
@@ -0,0 +1,99 @@
+package ch.epfl.pop.authentication
+
+import akka.http.scaladsl.model.StatusCodes.ClientError
+import akka.http.scaladsl.model.{AttributeKey, HttpRequest, HttpResponse, StatusCodes}
+import akka.http.scaladsl.server
+import akka.http.scaladsl.server.Directives._
+import ch.epfl.pop.authentication.Authenticate.verifyResponseType
+
+object Authenticate {
+  case class RequestParameters(response_type: String, client_id: String, redirect_uri: String, scope: String, state: Option[String], response_mode: Option[String], login_hint: String, nonce: String)
+
+  type VerificationState = Either[(String, String), Unit]
+
+  private val mandatoryParameters = Set("response_type", "client_id", "redirect_uri", "scope", "login_hint", "nonce")
+
+  def buildRoute(): server.Route = {
+    extractRequest { request =>
+      parameters(
+        "response_type",
+        "client_id",
+        "redirect_uri",
+        "scope",
+        "state".optional,
+        "response_mode".optional,
+        "login_hint",
+        "nonce"
+      ) {
+        (response_type, client_id, redirect_uri, scope, state, response_mode, login_hint, nonce) =>
```

Comment:
I think `client_id`,  `state`, `login_hint` and the `nonce` are left unchecked, wouldn't be a good idea to do so (if possible) or is that something bound to be done later maybe ?

Reply:
It is actually intended as they are left for the client (github for example using popcha for authentication) to use: client_id, state (optional) and nonce are entirely client-defined --> there are no ways to check them, neither any properties that we would like to enforce (nonce has a client-side property only). State actually should be a valid lao_id, but this doesn't need to be checked here as the user's pop app will do it to provide better flexibility, and we wouldn't actually have any means to check that now without heavily breaking the current encapsulation anyways (but again it is intended not to make that check here).